modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/RoleResourceV2Manager.java

```diff
@@ -328,6 +298,185 @@ public SCIMResponse updateWithPATCHRole(String id, String patchRequest, RoleV2Ma
         }
     }
 
+    @Override
```

Comment:
Why we implement these methods in RoleResourceV2Manager class instead in a new RoleResourceV3Manager class?

Reply:
If we can conclude that this class is a service class but not a class related to the API, then that would be fine.

Reply:
Since we’re not introducing any logic that explicitly ties this improvement to version 3, I believe it’s unnecessary to create a separate RoleResourceManager class solely for this change. Also, doing so would introduce unnecessary code duplication. IMO, we don't need to go with a `RoleResourceV3Manager` class